lib/Bugsnag.js

```diff
@@ -37,13 +36,10 @@ export class Client {
   handleUncaughtErrors = () => {
     if (ErrorUtils) {
       const previousHandler = ErrorUtils._globalHandler;
-      const bugsnag = this;
-
-      ErrorUtils.setGlobalHandler(function(error, isFatal) {
-        if (bugsnag.config.autoNotify) {
-          bugsnag.notify(error, function(report) {
-            report.severity = 'error';
-          });
+      
+      ErrorUtils.setGlobalHandler((error, isFatal) => {
+        if (this.config.autoNotify) {
+          this.notify(error, report => report.severity = 'error');
```

Comment:
I have a slight quibble with this line here. The implicit return is potentially problematic since the `notify` function uses the return value of this callback for it's own internal logic. 

Perhaps we should change it to `this.notify(error, report => { report.severity = 'error'});` to avoid future bugs getting introduced. 

Other than that I agree with the rest of these.